Cache/Requests.lean

```diff
@@ -8,6 +8,27 @@ import Cache.Hashing
 
 namespace Cache.Requests
 
+/-- Attempts to determine the running project's GitHub repository from its `origin` Git remote. -/
+def getRemoteRepo : IO String := do
+  let out ← IO.Process.output {cmd := "git", args := #["remote", "get-url", "origin"]}
+  unless out.exitCode == 0 do
+    throw <| IO.userError s!"\
+      Failed to run Git to determine project repository (exit code: {out.exitCode}).\n\
+      Ensure Git is installed and the `origin` remote points to the project's GitHub repository.\n\
+      Stdout:\n{out.stdout.trim}\nStderr:{out.stderr.trim}\n"
+  let url := out.stdout.trim.stripSuffix ".git"
+  let repo? : Option String := do
+    let pos ← url.revFind (· == '/')
+    let pos ← url.revFindAux (fun c => c == '/'  || c == ':') pos
+    return url.extract (url.next pos) url.endPos
```

Comment:
I think we should do stricter validation here. Make sure it looks exactly like a github git or https URL, with the right number of slashes, and whitelisted characters in between.

Reply:
Validation isn't that important here because the detected name is only being used by unprivileged users in `lake exe cache get`. When uploading to the cache with `put` in the CI, the repository should always be explicitly provided via `--repo`. I could make that more obvious by removing the default of `getRemoteRepo` entirely from `put` (replacing it with `MATHLIBREPO` for backwards compatibility).